operator-common/src/main/java/io/strimzi/operator/common/operator/resource/AbstractResourceDiff.java

```diff
@@ -24,6 +25,15 @@ protected JsonNode lookupPath(JsonNode source, String path) {
         return s;
     }
 
+    /**
+     * Empty node can have a value """" which compared with "" is different but semantically they are the same
+     * @param node tested node
+     * @return true if node is empty or has empty value
+     */
+    protected boolean nodeMissingOrEmpty(JsonNode node) {
```

Comment:
I'm not quite sure I understand this. `isMissingNode()` is the case where a property has the JSON value `null`. 
If you were conflating null with the empty string, I could understand, but I think you're trying to say that the string containing two double quotes is the same as the empty string  and the same as null. I don't understand why that is necessarily true. Remember, you're changing this for all JsonNodes in an STS, so while it _might_ be true for the `value` of a toleration, it might not be true elsewhere.

Reply:
The Kafka configuration mentioned above produces a STS like this 
```
      tolerations:
      - effect: NoSchedule
        key: kafkarole
        operator: Equal
```
note that `value` is missing and thus it is null although it is set to empty value in the Kafka CR.
I think it is kubernetes thing removing a property with no value.

Reply:
So why is the `.replace("\"", "")` necessary then?

Reply:
If we set `value: ''` it is propagated as `""`, in Java string it is `"\"\""` which is not equal to `""` (empty string).

Reply:
Per https://yaml.org/spec/1.2/spec.html#id2760844 YAML supports quoted scalars (using both `"` and `'`). So if `value: ''` results in the operator seeing the String with contents `""` I think that must be a bug somewhere (Jackson?).

Reply:
```
    @Test
    public void test() throws JsonProcessingException {
        JsonNode jsonNode = patchMapper().readTree("{\"effect\":\"NoSchedule\",\"value\":\"someValue\"}");
        System.out.println(jsonNode.path("value"));
    }
```
prints `"someValue"`. If we want just `someValue`, `asText()` method at the JsonNode object has to be called. 
I think it would be better to print `JsonNodeType.STRING` nodes without quotes.